### PR TITLE
executor: open doubleread close test and fix bug.

### PR DESCRIPTION
**Pull request by hanfei1991**
Open an old test and fix double read's bug.

### CODE REVIEW

File: executor/new_distsql.go [hunk 1]
```diff
@@ -295,6 +295,22 @@ func (e *IndexLookUpExecutor) executeTask(task *lookupTableTask, goCtx goctx.Con
...
+	defer cancel()
+	for {
+		select {
+		case task := <-workCh:
+			if task == nil {
+				return
+			}
+			e.executeTask(task, childCtx)
+		case <-childCtx.Done():
+		}
```
Comment (tiancaiamao) [inline executor/new_distsql.go]: return here

File: executor/new_distsql.go [hunk 2]
```diff
@@ -307,22 +323,11 @@ func (e *IndexLookUpExecutor) fetchHandlesAndStartWorkers() {
 	}()
 
 	lookupConcurrencyLimit := e.ctx.GetSessionVars().IndexLookupConcurrency
-	txnCtx := e.ctx.GoCtx()
 	for i := 0; i < lookupConcurrencyLimit; i++ {
-		go func() {
-			childCtx, cancel := goctx.WithCancel(txnCtx)
-			defer cancel()
-			select {
```
Comment (tiancaiamao) [inline executor/new_distsql.go]: the old code forget the for loop

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**